feat: Add Jacoco for code coverage and GitHub Actions workflow

khaledeid1k opened the pull request.

ZeyadAbdullah679 commented on presentation/build.gradle.kts:
```diff
@@ -30,9 +31,35 @@ android {
+        violationRules {
+            rule {
+                limit {
+                    minimum = BigDecimal("0.0")
```
Could you update that with the current limit required?

ZeyadAbdullah679 commented on design_system/build.gradle.kts:
```diff
@@ -34,12 +35,36 @@ android {
+        violationRules {
+            rule {
+                limit {
+                    minimum = BigDecimal("0.0")
```
Could you update that with the current limit required?

khaledeid1k committed:
feat: Add Jacoco for code coverage and GitHub Actions workflow

This commit introduces Jacoco for code coverage analysis in the `presentation`, `app`, and `design_system` modules.

It also adds a GitHub Actions workflow (`.github/workflows/CineVerse.yml`) to:
- Build the project
- Run unit tests
- Verify Jacoco coverage
- Generate and upload Jacoco HTML reports
- Comment on pull requests with coverage information

Jacoco configuration includes:
- Setting `toolVersion` to "0.8.13"
- Customizing the reports directory
- Adding `jacocoTestCoverageVerification` task with a minimum coverage limit
- Configuring JacocoTaskExtension for tests

Removed `androidx.navigation.compose` and `koin.androidx.compose` dependencies from `presentation/build.gradle.kts`.
Removed `coil.compose` and `coil.network.okhttp` dependencies from `design_system/build.gradle.kts`.

khaledeid1k committed:
Refactor: Streamline GitHub Actions workflow

- Renamed the 'build' job to 'compile' for clarity.
- Removed the 'Clean Project' and 'Compile Project' steps as 'assembleDebug' implicitly handles these.
- Standardized JDK version to 17 with 'temurin' distribution across jobs.
- Removed steps related to JaCoCo report generation and uploading, as well as commenting on coverage in pull requests. The workflow now focuses on verifying coverage via `jacocoTestCoverageVerification`.

ZeyadAbdullah679 committed:
Merge branch 'develop' into feature/setup-CI-CD
